pkg/channel/grpc/grpc_channel_test.go

```diff
@@ -13,6 +13,7 @@ import (
 	"net"
```

Comment:
@abserari Can you add a test in this file to assert that the context cancellation on timeout works as expected? add a similar test for readTimeout in the `http_channel_test` file?

Reply:
sure.